### PR TITLE
Search for IR.Function deeper in the IR tree to find instance methods

### LINKED ISSUE #6891: Runtime check of ascribed types works only for static methods
Try running:
```
from Standard.Base import all

f (x : Integer) = x+x


type My_Type
    Value
    
    g self (y : Integer) = y+y+y

main =
    o = My_Type.Value
    IO.println (o.g 1)
    IO.println (o.g "A")

    h (z : Integer) = z+z+z+z
    IO.println (h 1)
    IO.println (h "A")

    IO.println (f 1)
    IO.println (f "A")
```

### Expected behaviour:

It should fail with type error at `o.g "A"` and (if the first erroneous line is commented out) on `h "A"`.

Something like:
```
3
Execution finished with an error: Type error: expected `y` to be Integer, but got Text.
        at <enso> My_Type.g(test.enso:9:?-??)
        at <enso> test.main<arg-1>(test.enso:14:??-??)
        at <enso> test.main(test.enso:14:?-??)
```

### Actual behaviour

```
3
AAA
4
AAAA
2
Execution finished with an error: Type error: expected `x` to be Integer, but got Text.
        at <enso> test.f(test.enso:3:1-22)
        at <enso> test.main<arg-1>(test.enso:21:17-21)
        at <enso> test.main(test.enso:21:5-22)
```

The typechecks in `o.g` and `h` are not performed.

The local definition (`h`) is probably of a lower priority, as we don't need these as much - although ideally we want it to behave in a consistent way with other kinds of definitions - otherwise making the language behave inconsistently will make it harder to learn (as the user will need to know all the exceptions).

But the `o.g` case is very important - most of our library APIs are methods on types, we have very very few standalone top-level methods. So to be able to actually benefit from the ascribed typechecks, we need to ensure that they work correctly in such scenarios.

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Fixes #6891 by _going_ deeper into the IR tree to find more `IR.Function` instances.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

File: engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java [hunk 1]
```diff
@@ -95,6 +95,61 @@ public void runtimeCheckOfAscribedFunctionSignature() throws Exception {
...
+    type Neg
+        Singleton
+
+        twice self (a : Integer) = a + a
+    """, uri.getHost())
+            .uri(uri)
+            .buildLiteral();
+
+    var module = ctx.eval(src);
+    var neg = module.invokeMember("eval_expression", "Neg.Singleton.twice");
```
Comment (radeusgd) [inline engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java]: Shouldn't that be?

```suggestion
    var neg = module.invokeMember("eval_expression", "Neg.twice");
```

**Review by radeusgd — COMMENTED** [review]
Does this also fix the in-line case?

i.e.
```
foo x =
    bar (y : Integer) = y+y
    bar x
```

**Comment by JaroslavTulach** [comment]
> Does this also fix the in-line case?
> 
> i.e.
> 
> ```
> foo x =
>     bar (y : Integer) = y+y
>     bar x
> ```

Yes, it does: tested in [835e07b](https://github.com/enso-org/enso/pull/6902/commits/835e07ba9e057d1242dbcaf25fdd5cdd0537111b)

File: engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java]: Interesting. I'd bet I was fixing static method (as that one was always working). Corrected in [1d50504](https://github.com/enso-org/enso/pull/6902/commits/1d505041f0448b34e6f664184679ba3721b71dbd)

**Review by Akirathan — APPROVED** [review]

**Review by hubertp — APPROVED** [review]

**Pull request merged**